### ISSUE TITLE
How to run validation on multiple evaluation datasets simultaneously during Qwen2.5-VL-7B-Instruct fine-tuning?

### ISSUE THREAD

**Issue by 08D20088** [QwenLM/Qwen2.5-VL#1408]
I’m trying to fine-tune **Qwen2.5-VL-7B-Instruct** and want to use **two evaluation datasets** (eval\_dataset\_A, eval\_dataset\_B) to compute validation loss **during training**.
I referred to the official fine-tuning parameters here:
[https://github.com/QwenLM/Qwen2.5-VL/tree/main/qwen-vl-finetune](https://github.com/QwenLM/Qwen2.5-VL/tree/main/qwen-vl-finetune)

I added the following parameters:

```
--eval_dataset_use eval_dataset_A,eval_dataset_B
--eval_strategy steps
--eval_steps 2
```

My goal: use **both eval\_dataset\_A and eval\_dataset\_B** for validation loss calculation at each `eval_steps`.

In `qwen-vl-finetune/qwenvl/train/trainer.py`, there is the following code:

```python
if data_args.data_packing:
    data_module = make_supervised_data_module_packed(tokenizer=tokenizer, data_args=data_args)
else:
    data_module = make_supervised_data_module(tokenizer=tokenizer, data_args=data_args)

trainer = Trainer(
    model=model, processing_class=tokenizer, args=training_args, **data_module
)
```

How should I modify this so that `Trainer` evaluates on **two different evaluation datasets**?

If anyone has done something similar, could you share example code?
